test/test-expression-side-effect.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as assert from 'assert';
+import * as extend from 'extend';
+
+import {defaultConfig} from '../src/agent/config';
+import * as scanner from '../src/agent/io/scanner';
+import * as SourceMapper from '../src/agent/io/sourcemapper';
+import * as utils from '../src/agent/util/utils';
+import * as debugapi from '../src/agent/v8/debugapi';
+import {StatusMessage} from '../src/client/stackdriver/status-message';
+import {Common, LoggerOptions} from '../src/types/common';
+import * as stackdriver from '../src/types/stackdriver';
+
+const code = require('./test-expression-side-effect-code.js');
+
+const common: Common = require('@google-cloud/common');
+
+const itWithInspector = utils.satisfies(process.version, '>=10') ? it : it.skip;
```

Comment:
nit: add a comment that we use inspector as the default on 10+, hence the skip on earlier versions.